From 239e6b290b50ceb819399d858b117f1a876c4d71 Mon Sep 17 00:00:00 2001
From: Rob Herring <r.herring@freescale.com>
Date: Tue, 8 Apr 2008 13:33:06 -0500
Subject: [PATCH] ENGR00073706 ipu3: set axi id on high priority channels to 1.

This changes the high priority channels axi bus id to 1 allowing the
M4IF to give the channels high priority for bus arbitration.

Signed-off-by: Rob Herring <r.herring@freescale.com>
---
 drivers/mxc/ipu3/ipu_common.c    |    3 +++
 drivers/mxc/ipu3/ipu_param_mem.h |    5 +++++
 2 files changed, 8 insertions(+), 0 deletions(-)

diff --git a/drivers/mxc/ipu3/ipu_common.c b/drivers/mxc/ipu3/ipu_common.c
index d5aee68..10b532c 100644
--- a/drivers/mxc/ipu3/ipu_common.c
+++ b/drivers/mxc/ipu3/ipu_common.c
@@ -635,6 +635,9 @@ int32_t ipu_init_channel_buffer(ipu_channel_t channel, ipu_buffer_t type,
 	burst_size = _ipu_ch_param_get_burst_size(dma_chan);
 	_ipu_ic_idma_init(dma_chan, width, height, burst_size, rot_mode);
 
+	if (idma_is_set(IDMAC_CHA_PRI, dma_chan))
+		_ipu_ch_param_set_high_priority(dma_chan);
+
 	_ipu_ch_param_dump(dma_chan);
 
 	spin_lock_irqsave(&ipu_lock, lock_flags);
diff --git a/drivers/mxc/ipu3/ipu_param_mem.h b/drivers/mxc/ipu3/ipu_param_mem.h
index b520892..e416340 100644
--- a/drivers/mxc/ipu3/ipu_param_mem.h
+++ b/drivers/mxc/ipu3/ipu_param_mem.h
@@ -312,4 +312,9 @@ static inline void _ipu_ch_param_set_interlaced_scan(uint32_t ch)
 	ipu_ch_param_mod_field(ipu_ch_param_addr(ch), 1, 102, 14, stride - 1);
 };
 
+static inline void _ipu_ch_param_set_high_priority(uint32_t ch)
+{
+	ipu_ch_param_mod_field(ipu_ch_param_addr(ch), 1, 93, 2, 1);
+};
+
 #endif
-- 
1.5.4.4

